js/app.js

```diff
@@ -1,80 +1,131 @@
-      };
-
-      this.todos = [...this.todos, todo];
-      this.todoList.innerHTML =
-        this.getTodoElement(todo) + this.todoList.innerHTML;
-      this.todoInput.value = '';
-
-      this.saveTodos();
-    });
+  /**
+   * Creates a new todo.
+   *
+   * @memberof ToDoApp
+   */
+  createTodo = (todoName) => {
+    const todo = {
+      id: +new Date(),
+      name: todoName,
+      complete: false,
+    };
+
+    this.todos = [todo, ...this.todos];
+    this.renderTodos(this.todos);
+
+    this.saveTodosToLocalStorage();
+    this.todoInput.value = '';
   };
 
-  toggleCompletedEventHandler = () => {
-    this.todoList.addEventListener('toggleComplete', (e) => {
-      const todoElement = e.target;
-      const todoItemInArray = this.todos.find(
-        (todo) => todo.id === +todoElement.id
-      );
-      todoItemInArray.complete = !todoItemInArray.complete;
-      todoElement.setAttribute('complete', todoItemInArray.complete);
-
-      this.saveTodos();
-    });
+  /**
+   * Toggles a todo as complete or not complete.
+   *
+   * @memberof ToDoApp
+   */
+  toggleComplete = (element) => {
+    const todoItemInArray = this.todos.find((todo) => todo.id === +element.id);
+    todoItemInArray.complete = !todoItemInArray.complete;
+    element.setAttribute('complete', todoItemInArray.complete);
+
+    this.saveTodosToLocalStorage();
   };
 
-  deleteTodoEventHandler = () => {
-    this.todoList.addEventListener('deleteTodo', (e) => {
-      const todoElement = e.target;
-      const todos = this.todos.filter((todo) => todo.id !== +todoElement.id);
-      this.todos = todos;
-      this.todoList.removeChild(todoElement);
-
-      this.saveTodos();
-    });
+  /**
+   * Deletes a given todo.
+   *
+   * @memberof ToDoApp
+   */
+  deleteTodo = (element) => {
+    const todos = this.todos.filter((todo) => todo.id !== +element.id);
+    this.todos = todos;
+    this.todoList.removeChild(element);
+
+    this.saveTodosToLocalStorage();
   };
 
-  saveTodos = () => {
+  /**
+   * Save the todos to localStorage.
+   *
+   * @memberof ToDoApp
+   */
+  saveTodosToLocalStorage = () => {
     const todos = JSON.stringify(this.todos);
 
     localStorage.setItem('todos', todos);
   };
 
-  getTodos = () => {
+  /**
+   * Gets todo from localStorage.
+   *
+   * @memberof ToDoApp
+   * @returns {Array} todos
+   */
+  getTodosFromLocalStorage = () => {
     const todos = JSON.parse(localStorage.getItem('todos')) || [];
 
-    return todos.reverse();
+    return todos;
   };
 
-  renderTodos = () => {
-    const todos = this.todos.map((todo) => this.getTodoElement(todo));
-
-    this.todoList.innerHTML = [...todos].join('');
+  /**
+   * Renders all the todo items.
+   *
+   * @memberof ToDoApp
+   */
+  renderTodos = (todos) => {
+    const todosTemplate = html`
+      ${repeat(
+        todos,
+        (todos) => todos.id,
+        (todo) => this.getTodoElement(todo)
+      )}
+    `;
+
+    render(todosTemplate, this.todoList);
```

Comment:
lit-element has its own life cycle, like, render(), firstUpdated() etc, lets use that instead.